firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java

```diff
@@ -90,14 +92,14 @@ Object convertValue(Value value) {
     return result;
   }
 
-  private Object convertServerTimestamp(ServerTimestampValue value) {
+  private Object convertServerTimestamp(Value serverTimestampValue) {
     switch (serverTimestampBehavior) {
       case PREVIOUS:
-        return value.getPreviousValue() == null ? null : convertValue(value.getPreviousValue());
+        Value previousValue = getPreviousValue(serverTimestampValue);
+        if (previousValue == null) return null;
```

Comment:
The Java style guide suggests to use braces even for one line statements: https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s4.1-braces

Reply:
Fixed. It's lame that Java and C++ don't align on this point.